src/components/studies/scheduler/NotificationTime.tsx

```diff
@@ -114,6 +114,32 @@ const NotificationTime: React.FunctionComponent<NotificationTimeProps> = ({
     }
   }, [offset, isMultiday])
 
+  // Hook to track previous value of a prop, from: https://stackoverflow.com/a/57706747
+  const usePrevious = <T extends unknown>(value: T): T | undefined => {
+    const ref = React.useRef<T>()
+    React.useEffect(() => {
+      ref.current = value
+    })
+    return ref.current
+  }
+
+  /* Re-calculate offset when windowStartTime changes. 
+  Track prevWindowStartTime so can tell when windowStartTime has actually changed, as opposed to 
+  just when the offset is updated, which updates the NotificationWindow key and treats windowStarTime as changed. 
+  See: https://stackoverflow.com/a/60979832 */
+  const prevWindowStartTime = usePrevious(windowStartTime)
+  React.useEffect(() => {
+    if (
+      isFollowUp &&
+      isMultiday &&
+      prevWindowStartTime !== undefined &&
+      windowStartTime !== undefined &&
+      prevWindowStartTime !== windowStartTime
+    ) {
+      changeMultidayOffset({days: daysForMultidayOffset, time: timeForMultidayOffset})
+    }
+  }, [windowStartTime])
```

Comment:
I'm a little confused here. If this useEffect only runs when windowStartTime. changes, which it should, since that's what we have in dependency array, why do we need `prevWindowStartTime !== windowStartTime`

Reply:
I was confused too -- without `prevWindowStartTime !== windowStartTime`, this useEffect is triggered every time the offset is changed, even if windowStartTime is unchanged. 

I think it's because the key for NotificationWindow (parent of NotificationTime) includes the offset. So when the offset changes, a new NotificationWindow instance (containing a new NotificationTime instance) is mounted, so this useEffect is re-run when the new instance is rendered, even though windowStartTime hasn't changed. 

I found [these](https://stackoverflow.com/a/60979832) [two](https://stackoverflow.com/a/54924104) stackoverflow answers to be helpful while troubleshooting